src/main/java/com/onarandombox/MultiverseNetherPortals/listeners/MVNPPlayerListener.java

```diff
@@ -62,44 +57,44 @@ public void onPlayerPortal(PlayerPortalEvent event) {
             }
         }
 
+        Location newTo;
+        String currentWorld = currentLocation.getWorld().getName();
         String linkedWorld = this.plugin.getWorldLink(currentWorld, type);
-
-        if (linkedWorld != null) {
-            this.linkChecker.getNewTeleportLocation(event, currentLocation, linkedWorld);
+        if (currentWorld.equalsIgnoreCase(linkedWorld)) {
+            newTo = null;
+        } else if (linkedWorld != null) {
+            newTo = this.linkChecker.findNewTeleportLocation(currentLocation, linkedWorld, event.getPlayer());
         } else if (this.nameChecker.isValidNetherName(currentWorld)) {
             if (type == PortalType.NETHER) {
-                this.plugin.log(Level.FINER, "");
-                this.linkChecker.getNewTeleportLocation(event, currentLocation, this.nameChecker.getNormalName(currentWorld, PortalType.NETHER));
+                newTo = this.linkChecker.findNewTeleportLocation(currentLocation, this.nameChecker.getNormalName(currentWorld, PortalType.NETHER), event.getPlayer());
```

Comment:
What's the diff between these two calls?

Reply:
The difference is that `getNewTeleportLocation` uses `event.setTo()` to set the new location, but this caused issues where the new location could be set to `null` and cause an error (see #158). `findNewTeleportLocation` does the exact same thing except that it returns the value instead of setting it directly (then we can check if it's `null` before using it).